docs/architecture/adr-054-crypto-encoding-2.md

```diff
@@ -0,0 +1,85 @@
+# ADR 054: Crypto encoding (part 2)
+
+## Changelog
+
+\*2020-2-27: Created
+
+## Context
+
+Amino has been a pain point of many users in the ecosystem. While Tendermint does not suffer greatly from the performance degradation introduced by amino, we are making an effort in moving the encoding format to a widely adopted format, [Protocol Buffers](https://developers.google.com/protocol-buffers). With this migration a new standard is needed for the encoding of keys. This will cause ecosystem wide breaking changes.
+
+Currently amino encodes keys as `<PrefixBytes> <Length> <ByteArray>`.
+
+> With keys becoming a [oneof type](https://developers.google.com/protocol-buffers/docs/proto3#oneof) we have the option of removing the prefix bytes.
+
+## Decision
+
+Transitioning from a fixed size byte array to bytes would be the first step. This will enable usage of [cosmos-proto](https://github.com/regen-network/cosmos-proto) interface type. This removes boiler plate needed for oneof types.
```

Comment:
When you say "bytes," do you mean a byte slice?

Reply:
yes will detail that in the paragraph, thanks